hs_core/search_indexes.py

```diff
@@ -356,3 +378,166 @@ def prepare_editors_count(self, obj):
             return obj.raccess.edit_users.all().count()
         else:
             return 0
+
+    def prepare_geometry_type(self, obj):
+        if hasattr(obj, 'metadata'):
+            if isinstance(obj.metadata, GeographicFeatureMetaData):
+                return obj.metadata.geometryinformation.all().first().geometryType
```

Comment:
Do you want to me to check whether the object is GeographicFeatureResource first before access its GeographicFeatureMetaData? As @alvacouch suggested, instead of checking the type of the resource, we can actually check whether the object's metadata is the specific type we want to access. For this example, if the object's metadata is GeographicFeatureMetaData, we can have access to geometryType without exception.